include/libzutil.h

```diff
@@ -276,11 +276,15 @@ _LIBZUTIL_H void update_vdev_config_dev_sysfs_path(nvlist_t *nv,
  * Thread-safe strerror() for use in ZFS libraries
  */
 static inline char *zfs_strerror(int errnum) {
-#ifdef HAVE_STRERROR_L
-	return (strerror_l(errnum, uselocale(0)));
-#else
-	return (strerror(errnum));
-#endif
+	static __thread char errbuf[2048];
+	static __thread pthread_mutex_t zfs_strerror_lock =
+		PTHREAD_MUTEX_INITIALIZER;
```

Comment:
I doubt mutex declared as per-thread makes any sense.

Also may be questionable, but I wonder whether we need 2KB buffer here.  I'd guess 256-512 bytes should be more than enough for all standard errors in worst possible locales.

Reply:
My bad, sorry. That really does not make any sense :)

Reply:
Fixed mutex definition, reduced buffer size to 512 bytes, and if possible, use strerror_r(). That really should behave as original strerror(), howewer, without additional mutex locking.